### PR TITLE
Generate documentation using Docile and MkDocs

### PR DESCRIPTION
**Pull request by rsrock**
To build new documentation, run `julia docs/build.jl` then `mkdocs gh-deploy --clean` 
from within the Images dir.

Requires MkDocs

Supersedes #403

### CODE REVIEW

File: docs/src/function_reference.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+map!(mapi, dest, img)
+```
+
+can be used to specify both the form of the result and the algorithm used.
+
+# Image I/O
+
+See also the more [thorough description](extendingIO.html).
+
+TODO: edit to point to FileIO
```
Comment (timholy) [inline docs/src/function_reference.md]: TODO?

**Comment by timholy** [comment]
This looks awesome! I'll merge this and make the small tweak myself. Sorry I forgot about this.

**Comment by timholy** [comment]
Trying to run this locally, I got this:
```
$ mkdocs gh-deploy --clean
/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py:86: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see http://click.pocoo.org/python3/
  @common_options
/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py:100: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see http://click.pocoo.org/python3/
  @common_options
/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py:125: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see http://click.pocoo.org/python3/
  @common_options
/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py:145: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see http://click.pocoo.org/python3/
  @common_options
/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py:176: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see http://click.pocoo.org/python3/
  @common_options
/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py:194: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see http://click.pocoo.org/python3/
  @common_options
ERROR   -  Config value: 'docs_dir'. Error: The path docs/build doesn't exist 

Aborted with 1 Configuration Errors!
```
If I manually create `docs/build`, then I get this (this time suppressing all the warnings):
```
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /home/tim/.julia/v0.4/Images/site 
ERROR   -  file not found: /home/tim/.julia/v0.4/Images/docs/build/home.md 
ERROR   -  Error building page home.md 
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs/cli.py", line 185, in gh_deploy_command
    build.build(config, clean_site_dir=clean)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs/build.py", line 299, in build
    build_pages(config)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs/build.py", line 259, in build_pages
    dump_json)
  File "/usr/local/lib/python2.7/dist-packages/mkdocs/build.py", line 171, in _build_page
    input_content = io.open(input_path, 'r', encoding='utf-8').read()
IOError: [Errno 2] No such file or directory: '/home/tim/.julia/v0.4/Images/docs/build/home.md'
```
I ran this from the top-level Images directory.

**Comment by timholy** [comment]
I should add:
```
$ python --version
Python 2.7.6
```

**Comment by rsrock** [comment]
Did you `julia docs/build.jl` from the Images dir first?  That should populate Images/docs/build.

One important detail that I just remembered-- you need `Docile.jl`, and you also need to checkout the `rewrite` branch of `Docile` to run `docs/build.jl`.

**Comment by timholy** [comment]
Awesome, that works. Thanks!

Another question: other thank the link to github.io, presumably there's no reason that `home.md` needs to be different from the README? Could we use a symlink or something?

**Pull request merged**